gr-fec/lib/polar_common.cc

```diff
@@ -108,14 +108,13 @@ namespace gr {
         d_volk_temp = (unsigned char*) volk_malloc(sizeof(unsigned char) * block_size(), volk_get_alignment());
         d_volk_frozen_bit_mask = (unsigned char*) volk_malloc(sizeof(unsigned char) * block_size(), volk_get_alignment());
         d_volk_frozen_bits = (unsigned char*) volk_malloc(sizeof(unsigned char) * nfrozen, volk_get_alignment());
-        for(int i = 0; i < nfrozen; i++){
-          d_volk_frozen_bits[i] = d_frozen_bit_values[i];
-        }
+        std::copy(d_frozen_bit_values.begin(), d_frozen_bit_values.end(), d_volk_frozen_bits);
+        std::fill(d_volk_frozen_bits + d_frozen_bit_values.size(), d_volk_frozen_bits + nfrozen, 0);
 
         int nfbit = 0;
         for(int i = 0; i < block_size(); i++){
           unsigned char m = 0x00;
-          if(d_frozen_bit_positions[nfbit] == i){
+          if(nfbit < d_frozen_bit_positions.size() && d_frozen_bit_positions[nfbit] == i){
```

Comment:
I'd suggest to update this line to make it more C++ic.
It should state: `d_frozen_bit_positions.at(nfbit) == i`.
This would also throw an error if indices are incorrect.

Reply:
no, that's not more C++ic, at all; C++ lives its overloaded operators, and the fact that I enabled bounds checking on that (which `at` would do inherently, I agree) was what led us to this, but it's definitely not more C++ to use a method when you could use an index operator :)